Assets/SuperMultiplayerShooter/Scripts/PNManager.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Generic;
+using UnityEngine;
+using PubnubApi;
+using PubnubApi.Unity;
+
+public class PNManager : PNManagerBehaviour
+{
+    // UserId identifies this client.
+    public string userId;
+
+    //Persist the PubNub object across scenes
+    public static PNManager pubnubInstance;
+
+    //Cached players from connection.
+    private static Dictionary<string, UserMetadata> cachedPlayers = new Dictionary<string, UserMetadata>();

```

Comment:
FYI, I'll need an alternative to PhotonInstance.Name (I think it is, basically, the nickname the player has assigned themselves).  I plan on using this cachedPlayers dictionary.

Reply:
Sounds great! I've been using `pubnub.GetCurrentUserId()` and using that to compare to the cachedPlayers uuid to check if it's the player themselves.